llarp/path/transit_hop.cpp

```diff
@@ -260,9 +260,9 @@ namespace llarp
     {
       if (m_DownstreamQueue && not m_DownstreamQueue->empty())
       {
-        r->threadpool()->addJob([self = shared_from_this(),
-                                 data = std::move(m_DownstreamQueue),
-                                 r]() { self->DownstreamWork(data, r); });
+        r->QueueWork([self = shared_from_this(), data = std::move(m_DownstreamQueue), r]() {
+          self->DownstreamWork(data, r);
```

Comment:
Add a `mutable` to the lambda and then `std::move(data)` here to avoid the atomic ref increment